protected/humhub/components/InstallationState.php

```diff
@@ -0,0 +1,67 @@
+<?php
+
+namespace humhub\components;
+
+use humhub\helpers\DatabaseHelper;
+use Yii;
+use yii\base\BaseObject;
+use yii\base\StaticInstanceInterface;
+use yii\base\StaticInstanceTrait;
+
+class InstallationState extends BaseObject implements StaticInstanceInterface
+{
+    use StaticInstanceTrait;
+
+    public const STATE_NOT_INSTALLED = 1 << 0;
+    public const STATE_DATABASE_CONFIGURED = 1 << 1;
+    public const STATE_INSTALLED = 1 << 16;
+
+    private int $state;
+
+    public function init()
+    {
+        $this->state = Yii::$app->settings->get(self::class, self::STATE_NOT_INSTALLED);
+    }
+
+    public function setState(int $state): void
+    {
+        $this->state |= $state;
+
+        Yii::$app->settings->set(self::class, $this->state);
+    }
+
+    public function getState(): string
+    {
+        return $this->state;
+    }
+
+    public function hasState(int $state): bool
+    {
+        return ($this->state & $state) === $state;
+    }
+
+    public function isDatabaseConfigured()
+    {
+        return $this->hasState(self::STATE_DATABASE_CONFIGURED);
+    }
+
+    public function isDatabaseInstalled(): bool
+    {
+        $configExist = $this->hasState(self::STATE_DATABASE_CONFIGURED);
```

Comment:
installationState in singleton component, so it reading from database once for request, and after compares value from database comparing bits. It works faster than cache. Or you mean use cache instead of database?

Reply:
@gevorgmansuryan My idea was whether it is necessary to run the method  `Yii::$app->installationState->isDatabaseInstalled()` twice  in two places, we could remember the result.  Not really a cache, just a private class variable.

Reply:
@luke- yeah, currently it works like you mentioned,state is stored in InstallationState::$state variable, and InstallationState component itself has static instance